pubsub.go

```diff
@@ -268,3 +338,40 @@ func bootstrapPubsub(ctx context.Context, cr routing.ContentRouting, host p2phos
 
 	wg.Wait()
 }
+
+func (wg *watchGroup) add(ctx context.Context, outCh chan []byte) {
+	proxy := make(chan []byte, 1)
+	n := wg.n
+	wg.listeners[n] = proxy
+	wg.n++
+
+	go func() {
+		defer close(outCh)
+		defer func() {
+			wg.lk.Lock()
```

Comment:
There's a deadlock here if `notifyWatchers` is trying to write to our channel (it can write twice in a row so the buffer doesn't help). Unfortunately, the `watcher <- data` write may *also* have to select on the context (or at least the context cancellation channel). The map will probably need to be `map[chan <-[]byte]context.Context`.